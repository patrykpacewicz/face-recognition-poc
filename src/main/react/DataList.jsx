import React, { Component } from 'react';

import Paper from 'material-ui/Paper';
import {List, ListItem} from 'material-ui/List';
import Avatar from 'material-ui/Avatar';
import Subheader from 'material-ui/Subheader';

export default class DataList extends Component {
  render() {
    if (this.props.data.length <= 0) {
      return (<div />);
    }
    return (
      <Paper zDepth={3}>
        <List>
          <Subheader>{this.props.title}</Subheader>
          {this.props.data.map( (row, i) => (
              <ListItem key={i}
                leftAvatar={<Avatar src={row.imageUrl} />}
                primaryText={"Face: " + (i+1)}
                secondaryText={ <p> {JSON.stringify(row.position)} </p> }
                secondaryTextLines={2}
              />
          ))}
       </List>
     </Paper>
    );
  }
}

DataList.propTypes = {
  title: React.PropTypes.string,
  data: React.PropTypes.array
};
