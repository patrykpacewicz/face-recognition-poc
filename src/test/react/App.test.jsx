import React from 'react';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import DataList from '../../../src/main/react/DataList';

test('Link changes the class when hovered', () => {
    const component = renderer.create(
        <MuiThemeProvider>
            <DataList data={[{position: "", imageUrl: ""}]} title=""/>
        </MuiThemeProvider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});