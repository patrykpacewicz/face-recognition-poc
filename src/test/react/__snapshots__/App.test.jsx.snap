// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<div
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#ffffff",
      "borderRadius": 2,
      "boxShadow": "0 10px 30px rgba(0, 0, 0, 0.19),
             0 6px 10px rgba(0, 0, 0, 0.23)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "fontFamily": "Roboto, sans-serif",
      "muiPrepared": true,
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
    }
  }
>
  <div
    style={
      Object {
        "muiPrepared": true,
        "padding": "0px 0px 8px 0px",
      }
    }
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.54)",
          "fontSize": 14,
          "fontWeight": 500,
          "lineHeight": "48px",
          "muiPrepared": true,
          "paddingLeft": 16,
          "width": "100%",
        }
      }
    >
      
    </div>
    <div>
      <span
        disabled={undefined}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        open={null}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "background": "none",
            "backgroundColor": null,
            "border": 10,
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "cursor": "pointer",
            "display": "block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 16,
            "fontWeight": "inherit",
            "lineHeight": "16px",
            "margin": 0,
            "muiPrepared": true,
            "outline": "none",
            "padding": 0,
            "position": "relative",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "verticalAlign": null,
            "zIndex": 1,
          }
        }
        tabIndex={0}
      >
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
        >
          <div
            style={
              Object {
                "marginLeft": 0,
                "muiPrepared": true,
                "paddingBottom": 16,
                "paddingLeft": 72,
                "paddingRight": 16,
                "paddingTop": 16,
                "position": "relative",
              }
            }
          >
            <div
              className={undefined}
              size={40}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgb(188, 188, 188)",
                  "borderRadius": "50%",
                  "color": "#ffffff",
                  "display": "inline-flex",
                  "fontSize": 20,
                  "height": 40,
                  "justifyContent": "center",
                  "left": 16,
                  "muiPrepared": true,
                  "position": "absolute",
                  "top": 16,
                  "userSelect": "none",
                  "width": 40,
                }
              }
            />
            <div
              style={
                Object {
                  "muiPrepared": true,
                }
              }
            >
              Face: 1
            </div>
            <p
              style={
                Object {
                  "WebkitBoxOrient": "vertical",
                  "WebkitLineClamp": 2,
                  "color": "rgba(0, 0, 0, 0.54)",
                  "display": "-webkit-box",
                  "fontSize": 14,
                  "height": 36,
                  "lineHeight": "18px",
                  "margin": 0,
                  "marginTop": 4,
                  "muiPrepared": true,
                  "overflow": "hidden",
                  "textOverflow": "ellipsis",
                  "whiteSpace": null,
                }
              }
            >
               
              ""
               
            </p>
          </div>
        </div>
      </span>
    </div>
  </div>
</div>
`;
